src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -60,6 +60,8 @@ PDPServer2::PDPServer2(
 
 PDPServer2::~PDPServer2()
 {
+    discovery_db_.disable();
```

Comment:
before disabled the DDB, the PDPServer must stop the routine thread (`cancel_timer()`)